src/EditorFeatures/CSharpTest/AssignOutParameters/AssignOutParametersAtStartTests.cs

```diff
@@ -0,0 +1,632 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.AssignOutParameters;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AssignOutParameters
+{
+    public partial class AssignOutParametersAtStartTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new AssignOutParametersAtStartCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSimpleReturn()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSwitchSectionReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        switch (0)
+        {
+            default:
+                [|return 'a';|]
+        }
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        switch (0)
+        {
+            default:
+                return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingWhenVariableAssigned()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+            i = 1;
+
+        [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        i = 0;
+        if (b)
+            i = 1;
+
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths2()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    bool M(out int i1, out int i2)
+    {
+        [|return Try(out i1) || Try(out i2);|]
```

Comment:
>return [](start = 10, length = 6)

Shouldn't we expect an assignment to `i2` to be inserted before?

Reply:
@gafter I believe Cyrus bails out from this code fixer if we are on the first statement of the method, as the other code fixer (assign out parameters before the current statement) will provide the exact same fix. This will avoid identical duplicate code fixes. @CyrusNajmabadi to confirm.

Reply:
That is correct.  You'll see this test in AssignOutParametersAboveReturnTests.cs, where it does assign to i2: https://github.com/dotnet/roslyn/pull/37395/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.resx&file-filters%5B%5D=.vb&file-filters%5B%5D=.xlf#diff-b89984a32a497a4c26cae919050298d6R127-R131